### PR TITLE
Report elapsed time

### PR DESCRIPTION
**Pull request by g-arjones**

### CODE REVIEW

File: lib/autoproj/reporter.rb [hunk 1]
```diff
@@ -49,8 +49,25 @@ def error(error)
...
         end
+
+        def reset_timer
+            @timer_start = Time.now
+        end
+
+        def elapsed_time
+            return nil unless @timer_start
+            secs = Time.now - @timer_start
+            [[60, :seconds], [60, :minutes], [24, :hours], [1000, :days]].map do |count, name|
```
Comment (doudou) [inline lib/autoproj/reporter.rb]: I think you might want to read more about [Time#strftime](https://ruby-doc.org/core-2.2.0/Time.html#method-i-strftime)

Comment (doudou) [reply lib/autoproj/reporter.rb]: Right ... you can't if yout want to achieve the same result (that is, not showing the time periods that are irrelevant). Sorry for the noise.

Comment (g-arjones) [reply lib/autoproj/reporter.rb]: No problem... :+1:

File: lib/autoproj/reporter.rb [hunk 2]
```diff
@@ -49,8 +49,28 @@ def error(error)
...
+                    secs, n = secs.divmod(count)
+                    next if (val = n.to_i) == 0
+                    "#{val} #{val > 1 ? name + 's' : name}"
+                end
+            end.compact.reverse.join(' ')
+        end
+
         def success
-            Autoproj.message("Command finished successfully at #{Time.now}", :bold, :green)
+            elapsed_string = elapsed_time ? " (took #{elapsed_time})" : ''
```
Comment (doudou) [inline lib/autoproj/reporter.rb]: nil.to_s is the emptry string. (and ?: is very much un-rubyish). A better way (as in more rubocopyish) would be to just assign ... if 

~~~
elapsed_string = " (took #{elapsed_time})" if elapsed_time
~~~

Comment (doudou) [reply lib/autoproj/reporter.rb]: (no need to change, just for future reference)

Comment (g-arjones) [reply lib/autoproj/reporter.rb]: Hmm.. This is new to me.

```ruby
if false
    foo = 'bar'
end
```

I thought `foo` would be unset, in which case `puts foo` would raise `NoMethodError` but Ruby conveniently sets `foo` to `nil` even though the expression is false. Pretty neat :+1:

Comment (doudou) [reply lib/autoproj/reporter.rb]: In Ruby, local variables are defined after their first assignation, i.e. they are defined (and initialized to `nil`) in *all lines of the same scope* as soon as the parser sees them assigned.

The rule sounds really weird, but it actually makes a lot of sense in the code.

Comment (g-arjones) [reply lib/autoproj/reporter.rb]: > makes a lot of sense in the code.

Definitely. Thanks!

**Pull request merged**